src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -991,20 +1005,18 @@ EclipseGrid::EclipseGrid(const Deck& deck, const int * actnum)
             CoordMapper cm(this->getNX(), this->getNY());
             std::vector<double> zcorn( zm.size() );
             std::vector<double> coord( cm.size() );
-            {
-                std::vector<double> zk(this->getNZ());
-                zk[0] = 0;
-                for (std::size_t k = 1; k < this->getNZ(); k++)
-                    zk[k] = zk[k - 1] + dzv[k - 1];
-
+            {                    
+                std::vector<double> zdepth(tops);
+                auto z = dz.begin();
                 for (std::size_t k = 0; k < this->getNZ(); k++) {
+                    auto d = zdepth.begin();
                     for (std::size_t j = 0; j < this->getNY(); j++) {
                         for (std::size_t i = 0; i < this->getNX(); i++) {
-                            size_t tops_value = tops[ i + this->getNX() * j];
                             for (size_t c=0; c < 4; c++) {
-                                zcorn[ zm.index(i,j,k,c) ]     = zk[k] + tops_value;
-                                zcorn[ zm.index(i,j,k,c + 4) ] = zk[k] + tops_value + dzv[k];
+                                zcorn[ zm.index(i,j,k,c) ]     = *d;
```

Comment:
No - this should be `zdepth[k]` and `z[g]` with `g = i + j*nx + k*nx*ny`

Reply:
No problem, is that a general approach in the code base? Prefer indexing over pointer-like arithmetic?

Reply:
Depends on who you ask I guess - personally I tend to avoid pointer arithmetic. For this particular code the preference is at least very clear in favor of `[]`